### PR TITLE
Add publisher support for Algeron Addon

### PR DESCRIPTION
**Pull request by lordofthejars**
Add publisher support for Algeron Addon

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by lordofthejars** [comment]
Finally using the project configuration I have been able to make something persistent because Facets are not stateful per se, they are recreated as you mention some time ago.

Now I need to see why isInstalled is not called in Mac

**Comment by MatousJobanek** [comment]
good job 👍

File: src/main/java/org/jboss/forge/arquillian/api/ArquillianConfig.java [hunk 1]
```diff
@@ -18,7 +18,26 @@ public ArquillianConfig() {
    public ArquillianConfig(InputStream is) {
       xml = XMLParser.parse(is);
    }
-   
+
+   public boolean addExtension(String qualifier)
+   {
+      Node containerConfig = xml.getSingle("extension@qualifier=" + qualifier);
```
Comment (bartoszmajsak) [inline src/main/java/org/jboss/forge/arquillian/api/ArquillianConfig.java:24]: Can we extract `extension@qualifier=` to constant?

File: src/main/java/org/jboss/forge/arquillian/api/algeron/AlgeronPublisherFacet.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import org.jboss.forge.arquillian.api.ArquillianConfig;
+import org.jboss.forge.arquillian.api.ArquillianFacet;
+import org.jboss.forge.arquillian.api.YamlGenerator;
+import org.jboss.forge.arquillian.testframework.algeron.AlgeronConsumer;
+
+import java.util.Map;
+
+@FacetConstraint(ArquillianFacet.class)
+@FacetConstraint(AlgeronConsumer.class)
+@FacetConstraint(DependencyFacet.class)
```
Comment (bartoszmajsak) [inline src/main/java/org/jboss/forge/arquillian/api/algeron/AlgeronPublisherFacet.java:18]: I don't think `DependencyFacet` is necessary to be defined here, as it's implicitly required by `ArquillianFacet`

**Comment by lordofthejars** [comment]
@bartoszmajsak I need this to be merge to continue my work, since @gastaldi and @MatousJobanek give the ok, I am going to merge it. We can always improve code if something is found in next PRs.

**Comment by bartoszmajsak** [comment]
Sure, but changes I asked for were 5 minutes changes.

**Comment by bartoszmajsak** [comment]
Ok, just noticed they were not published 😂

**Comment by lordofthejars** [comment]
Yes hehehe, I didn't see anything so I merged

**Review by bartoszmajsak — COMMENTED** [review]
If you can bring those directly to master it would be awesome @lordofthejars

**Pull request merged**